src/site/_filters/i18n.js

```diff
@@ -60,6 +60,24 @@ const walk = (dir) => {
 // Scan the _data/i18n directory and create a nested i18n data object
 const data = {i18n: walk(path.join(__dirname, '..', '_data', 'i18n'))};
 
+/**
+ * Flattens the i18n yaml object to contain only translations from given locale.
+ * @param {Object} obj Object with i18n translations (from _data/i18n).
+ * @param {string} locale A locale prefix (example: 'en', 'pl')
+ * @return {Object|string}
+ */
+const getTranslation = function (obj, locale) {
+  const out = {};
+  for (const prop in obj) {
+    if (typeof obj[prop] === 'string') {
+      return obj[locale] ?? obj[defaultLocale];
+    } else {
+      out[prop] = getTranslation(obj[prop], locale);
+    }
+  }
+  return out;
+};
+
 /**
  * Looks for the i18n string that matches the path and locale.
```

Comment:
This method now returns either a group of messages or an individual string.

Is it worth making two methods since the expected results are so different - we can throw in either case?